tests/route/test_default_route.py

```diff
@@ -62,3 +145,38 @@ def test_default_ipv6_route_next_hop_global_address(duthosts, enum_rand_one_per_
     for nh in rtinfo['nexthops']:
         pytest_assert(not nh[0].is_link_local, \
                 "use link local address {} for nexthop".format(nh[0]))
+
+
+def test_default_route_with_bgp_flap(duthosts, enum_rand_one_per_hwsku_frontend_hostname, enum_asic_index, tbinfo):
+    """
+    Check the default route present in app_db has the correct nexthops ip
+    Check the default route is removed when the bgp sessions are shutdown
+     
+    """
+
+    pytest_require('t1-backend' not in tbinfo['topo']['name'], \
+            "Skip this testcase since this topology {} has no default routes"\
+                .format(tbinfo['topo']['name']))
+
+    duthost = duthosts[enum_rand_one_per_hwsku_frontend_hostname]
+    asichost = duthost.asic_instance(enum_asic_index)
+    af_list = ['ipv4', 'ipv6']
+
+    # verify the default route is correct in the app db
+    for af in af_list:
+        verify_default_route_in_app_db(asichost, tbinfo, af)
+
+    duthost.command("sudo config bgp shutdown all")
+    if not wait_until(120, 2, 0, duthost.is_bgp_state_idle):
+        pytest.fail(
+            'BGP Shutdown Timeout: BGP sessions not shutdown after 120 seconds')
+
+    # give some more time for default route to be removed
+    time.sleep(120)
```

Comment:
how about replacing this sleep with `wait_until` to constantly check the default route removal?

Reply:
address in latest commit